### PR TITLE
Add publishing app builds to saucelabs

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,18 +10,62 @@ on:
...
 
       - name: Run test
-        run: |
-          npm run test
+        run: npm run test
+
+      - name: Install JDK 13
+        uses: actions/setup-java@v1
+        with:
+          java-version: 13
```
Comment (valfirst) [inline .github/workflows/ci.yml:38]: why not 14?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -10,18 +10,62 @@ on:
...
+
+      - name: Gradle cache
```
Comment (valfirst) [inline .github/workflows/ci.yml:40]: group all cache actions

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -10,18 +10,62 @@ on:
...
+      
+      - name: Cocoapods cache
+        uses: actions/cache@v2
+        with:
+          path: ios/Pods
+          key: macos-pods-${{ hashFiles('**/Podfile.lock') }}
+          restore-keys: |
+            macos-pods-
+
+      - name: iOS build
```
Comment (valfirst) [inline .github/workflows/ci.yml:67]: why not to have 2 separate pipelines: for Android and iOS?

**Review by valfirst — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (uarlouski) [reply .github/workflows/ci.yml]: using the same version of java as locally

**Review by valfirst — CHANGES_REQUESTED** [review]

File: .github/workflows/android-ci.yml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      - name: Run lint
+        run: npm run lint
+
+      - name: Run test
+        run: npm run test
+
+      - name: Install JDK 13
+        uses: actions/setup-java@v1
+        with:
+          java-version: 13
```
[the remark this answers is not in the document]
Comment (uarlouski) [reply .github/workflows/android-ci.yml:48]: Starting a Gradle Daemon (subsequent builds will be faster)
265
java.lang.NoClassDefFoundError: Could not initialize class org.codehaus.groovy.vmplugin.v7.Java7
266
	at org.codehaus.groovy.vmplugin.VMPluginFactory.<clinit>(VMPluginFactory.java:43)
267
	at org.codehaus.groovy.reflection.GroovyClassValueFactory.<clinit>(GroovyClassValueFactory.java:35)
268
	at org.codehaus.groovy.reflection.ClassInfo.<clinit>(ClassInfo.java:107)
269
	at org.codehaus.groovy.reflection.ReflectionCache.getCachedClass(ReflectionCache.java:95)

[the remark this answers is not in the document]
Comment (uarlouski) [reply .github/workflows/android-ci.yml:48]: I'll update Gradle version separately

[the remark this answers is not in the document]
Comment (uarlouski) [reply .github/workflows/android-ci.yml:48]: https://github.com/vividus-framework/vividus-test-app/pull/5

**Review by valfirst — APPROVED** [review]

File: .github/workflows/android-ci.yml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+      - name: Install bundler
+        run: |
+          sudo gem install bundler
+          sudo bundle update --bundler
+
+      - name: Android build
+        env:
+          SAUCELABS_USER: ${{ secrets.SAUCELABS_USER }}
+          SAUCELABS_KEY: ${{ secrets.SAUCELABS_KEY }}
+        run: npm run build-android
```
Comment (valfirst) [inline .github/workflows/android-ci.yml:59]: only builds from `master` branch should be published to SL, PRs shouldn't publish anything

File: .github/workflows/ios-ci.yml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+      - name: Install bundler
+        run: |
+          sudo bundle update --bundler
+          sudo bundle install
+
+      - name: iOS build
+        env:
+          SAUCELABS_USER: ${{ secrets.SAUCELABS_USER }}
+          SAUCELABS_KEY: ${{ secrets.SAUCELABS_KEY }}
+        run: npm run build-ios
```
Comment (valfirst) [inline .github/workflows/ios-ci.yml:55]: only builds from `master` branch should be published to SL, PRs shouldn't publish anything

**Review by valfirst — CHANGES_REQUESTED** [review]

File: .github/workflows/android-ci.yml (hunk 5 above)
Comment (uarlouski) [reply .github/workflows/android-ci.yml]: I expected only me contributing this repo :)

File: .github/workflows/ios-ci.yml [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+      - name: iOS build
+        env:
+          SAUCELABS_USER: ${{ secrets.SAUCELABS_USER }}
+          SAUCELABS_KEY: ${{ secrets.SAUCELABS_KEY }}
+        run: |
+          PUBLISH_APP=false
+          if [[ $GITHUB_HEAD_REF = 'master' ]]; then
+            PUBLISH_APP=true
+          fi
```
Comment (valfirst) [inline .github/workflows/ios-ci.yml:59]: `PUBLISH_APP = [[ $GITHUB_HEAD_REF = 'master' ]]` - does it work?

Comment (uarlouski) [reply .github/workflows/ios-ci.yml:59]: it will not work because conditional operations in bash operate using status codes

**Review by valfirst — APPROVED** [review]

**Pull request merged**